src/Avalonia.Controls/Window.cs

```diff
@@ -62,6 +62,25 @@ public enum SystemDecorations
         Full = 2
     }
 
+    /// <summary>
+    /// Describes how the <see cref="Window.Closing"/> event behaves in the presence of child windows.
+    /// </summary>
+    public enum WindowClosingBehavior
+    {
+        /// <summary>
+        /// When the owner window is closed, the child windows' <see cref="Window.Closing"/> event
+        /// will be raised, followed by the owner window's <see cref="Window.Closing"/> events. A child
+        /// canceling the close will result in the owner Window's close being cancelled.
+        /// </summary>
+        OwnerAndChildWindows,
+
+        /// <summary>
+        /// When the owner window is closed, only the owner window's <see cref="Window.Closing"/> event
+        /// will be raised. This behavior is the same as WPF's.
+        /// </summary>
+        OwnerWindowOnly,
```

Comment:
I think the `Closing` and `Closed` events are a bit mixed here, at least for `OwnerAndChildWindows`. According to the PR description, it doesn't raise the `Closing` event for child windows. Is that right?

Reply:
When set to `OwnerAndChildWindows` which is the default and is the current behavior as of master, `Closing` will be raised on all child windows before being raised on the parent. `Closed` event is raised at the point that it's decided the window will be closed, i.e. all `Closing` handlers that were raised didn't cancel `Closing` event.